src/main/scala/robosky/ether/client/gui/GuiRegistry.scala

```diff
@@ -0,0 +1,32 @@
+package robosky.ether.client.gui
+
+import io.github.cottonmc.cotton.gui.CottonScreenController
+import io.github.cottonmc.cotton.gui.client.CottonScreen
+
+import net.fabricmc.fabric.api.client.screen.ScreenProviderRegistry
+
+import net.minecraft.container.BlockContext
+import net.minecraft.entity.player.{PlayerEntity, PlayerInventory}
+
+import robosky.ether.UplandsMod
+import robosky.ether.block.machine.infuser.InfuserContainer
+
+// delete this class?
+class InfuserScreen(container: InfuserContainer, player: PlayerEntity)
+  extends CottonScreen[InfuserContainer](container, player)
+
+object GuiRegistry {
+
+  val agisaltInfuserScreen = registerGui("aegisalt_infuser", new InfuserContainer(_, _, _), new InfuserScreen(_, _))
```

Comment:
get the container creation from the machine registry code

Reply:
Changed to retrieve the controller and the id from the machine entry